clients/apple/iOS/Widgets/DrawerView.swift

```diff
@@ -216,10 +208,105 @@ struct DrawerView<Main: View, Side: View>: View {
     }
 }
 
+// MARK: - DragValue
+struct DragValue {
+    let location: CGPoint
+    let startLocation: CGPoint
+    let translation: CGSize
+    let velocity: CGVector
+}
+
+// MARK: - NonCancellingDragGesture
+/// Like a DragGesture, but with `cancelsTouchesInView = false`
+struct NonCancellingDragGesture: UIGestureRecognizerRepresentable {
+    typealias UIGestureRecognizerType = UIPanGestureRecognizer
+
+    let minimumDistance: CGFloat
+    let onChanged: (DragValue) -> Void
+    let onEnded: (DragValue) -> Void
+
+    func makeCoordinator(converter: CoordinateSpaceConverter) -> Coordinator {
+        Coordinator(
+            minimumDistance: minimumDistance,
+            onChanged: onChanged,
+            onEnded: onEnded
+        )
+    }
+
+    @MainActor func makeUIGestureRecognizer(context: Context) -> UIGestureRecognizerType {
+        let recognizer = UIPanGestureRecognizer()
+        recognizer.cancelsTouchesInView = false // <- the only meaningful line in this struct
```

Comment:
I would argue we want this gesture to cancel touches in the view.

Wondering whether we are experiencing this now - someone goes to drag the sidebar open, then move vertically a bit, and the entire editor scrolls up. That is undesirable imo

Reply:
It's true, we are duplicating the inputs now to the drawer gesture and to workspace, and you can scroll the editor (or draw on a canvas) while opening the drawer. I'll fix this before merging and solicit another review from you